src/drawer/drawer.styles.tsx

```diff
@@ -73,26 +73,28 @@ export const Container = styled.div<StyleProps>`
 
 export const Header = styled.div`
     top: 0;
-
     display: flex;
     align-items: center;
     gap: ${Spacing["spacing-16"]};
     padding: ${Spacing["spacing-32"]} ${Spacing["spacing-16"]}
-        ${Spacing["spacing-16"]};
+        ${Spacing["spacing-16"]} ${Spacing["spacing-64"]};
```

Comment:
spacing is in px while the icon is in rem, using only the spacing token here will lead to misalignment when a different font size is set by the user

you can set `calc(<icon width> + <Spacing token>)` instead for more accurate positioning